src/PostgREST/Request/Types.hs

```diff
@@ -225,3 +227,8 @@ data FtsOperator
   | FilterFtsPhrase
   | FilterFtsWebsearch
   deriving Eq
+
+-- | Operators for filtering using the request body
+data BodyOperator
+  = BodyOpEqual
+  deriving Eq
```

Comment:
I think it's fine to have this as a special case for now, we can add underscore versions of the other operators later.

One question though, will this work for DELETE and RPC as well?

Reply:
It does not work for DELETE in the PR because the body is ignored, but it should work with the changes in #2355 (unless those changes are done here to avoid breaking changes in the future). 

RPC are a different story in which the filter of the result set is done outside of the `pgrst_source` CTE in the QueryBuilder.hs file, so it won't work out of the box unless the filter is done inside the CTE (may break some things). The resulting query  looks something like:

```sql
WITH pgrst_source AS (
  WITH pgrst_payload AS (...).
            pgrst_body AS (...),
            pgrst_args AS (...)
           SELECT ... -- The WHERE should be done here for the filter to get the data from pgrst_body
SELECT ...
FROM
  (SELECT "table".*
   FROM "pgrst_source" AS "table"
   WHERE "table"."id" = "pgrst_recordset_body"."id") _postgrest_t  -- it is done outside instead, which gives an error
```